src/Storages/NamedCollectionsHelpers.cpp

```diff
@@ -74,10 +72,21 @@ MutableNamedCollectionPtr tryGetNamedCollectionWithOverrides(
 
     NamedCollectionUtils::loadIfNot();
 
-    auto collection = tryGetNamedCollectionFromASTs(asts, throw_unknown_collection);
+    auto collection_name = getCollectionName(asts);
+    if (!collection_name.has_value())
+        return nullptr;
+
+    NamedCollectionPtr collection;
+    if (throw_unknown_collection)
+        collection = NamedCollectionFactory::instance().get(*collection_name);
+    else
+        collection = NamedCollectionFactory::instance().tryGet(*collection_name);
+
     if (!collection)
         return nullptr;
 
+    context->checkAccess(AccessType::USE_NAMED_COLLECTION, *collection_name);
```

Comment:
Probably the check should be performed even if the named collection doesn't exist. An user with no access rights is better not to be able to determine which named collections exist.